ql/src/java/org/apache/hadoop/hive/ql/plan/ImportTableDesc.java

```diff
@@ -24,6 +24,7 @@
 import java.util.Map;
 
 import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.metastore.TableType;
```

Comment:
The isExternal flag input during creation of CreateTableDesc should be set correctly.

Reply:
I am not able to see relation between the code changes and the comment. Can you please clarify?

Reply:
In the constructor, ImportTableDesc(String dbName, Table table), the creation of CreateTableDesc object takes 3rd argument as isExternal flag. Currently, it is passed false always. It should be set based on source table type.
Due to this, createTableDesc.isExternal() returns incorrect value for external tables.

Reply:
The reason I didn't touch that part was the comment about IMPORT statement as copied below

        this.createTblDesc = new CreateTableDesc(dbName,
                table.getTableName(),
                false, // isExternal: set to false here, can be overwritten by the IMPORT stmt

Looking at it again, it seems OK to set the flag based on the input table type and let IMPORT overwrite if needed.